Gems/ROS2/Code/Source/SimulationUtils/FollowingCameraComponent.cpp

```diff
@@ -170,93 +194,323 @@ namespace ROS2
+    {
+        AZ::Matrix4x4 localCamPoseChange = AZ::Matrix4x4::CreateIdentity();
+        float translationSpeed = m_configuration.m_translationSpeed;
+
+        // Check each key state and adjust the camera pose change accordingly
+        if (m_keyStates[AzFramework::InputDeviceKeyboard::Key::AlphanumericW])
         {
-            m_rotationOffset -= m_configuration.m_rotationSpeed;
-            return;
+            localCamPoseChange = localCamPoseChange * AZ::Matrix4x4::CreateTranslation(AZ::Vector3::CreateAxisY() * translationSpeed);
         }
-        if (channelId == AzFramework::InputDeviceKeyboard::Key::AlphanumericD)
+        if (m_keyStates[AzFramework::InputDeviceKeyboard::Key::AlphanumericS])
         {
-            m_rotationOffset += m_configuration.m_rotationSpeed;
-            return;
+            localCamPoseChange = localCamPoseChange * AZ::Matrix4x4::CreateTranslation(AZ::Vector3::CreateAxisY() * -translationSpeed);
+        }
+        if (m_keyStates[AzFramework::InputDeviceKeyboard::Key::AlphanumericA])
+        {
+            localCamPoseChange = localCamPoseChange * AZ::Matrix4x4::CreateTranslation(AZ::Vector3::CreateAxisX() * -translationSpeed);
         }
+        if (m_keyStates[AzFramework::InputDeviceKeyboard::Key::AlphanumericD])
+        {
+            localCamPoseChange = localCamPoseChange * AZ::Matrix4x4::CreateTranslation(AZ::Vector3::CreateAxisX() * translationSpeed);
+        }
+
+        m_cameraOffset = m_cameraOffset * localCamPoseChange;
+    }
 
-        // channelId is a numeric key (Key::Alphanumeric0-Key::Alphanumeric9)
-        if (auto it = KeysToView.find(channelId); it != KeysToView.end())
+    void FollowingCameraComponent::RotateCameraOnKeys()
+    {
+        const float rotationSpeed = m_configuration.m_rotationSpeed; // Reduce rotation speed for keys
+
+        float yawRotation = 0.0f; // Initialize yaw rotation to 0
+
+        if (m_keyStates[AzFramework::InputDeviceKeyboard::Key::AlphanumericQ])
         {
-            if (int viewId = it->second; viewId < m_configuration.m_predefinedViews.size())
+            yawRotation = rotationSpeed; // Rotate left
+        }
+        else if (m_keyStates[AzFramework::InputDeviceKeyboard::Key::AlphanumericE])
+        {
+            yawRotation = -rotationSpeed; // Rotate right
+        }
+
+        if (yawRotation != 0.0f) // Only proceed if there's a rotation to apply
+        {
+            // Fetch the current camera orientation from m_cameraOffset
+            AZ::Quaternion currentRotation = AZ::Quaternion::CreateFromMatrix4x4(m_cameraOffset);
+
+            // Global Z-axis for yaw rotation
+            AZ::Vector3 globalZAxis = AZ::Vector3(0, 0, 1);
+            AZ::Quaternion yawQuat = AZ::Quaternion::CreateFromAxisAngle(globalZAxis, yawRotation);
+
+            // Apply yaw rotation to current camera orientation
+            AZ::Quaternion newRotation = yawQuat * currentRotation;
+            newRotation.Normalize();
```

Comment:
No need to normalize.
assert on not normalized.

Reply:
Right normalization not needed - removed